src/core/server/server.ts

```diff
@@ -215,6 +217,8 @@ export class Server {
       coreId,
       'core',
       async (context, req, res): Promise<RequestHandlerContext['core']> => {
+        // TODO: this is exposed from start now. will need to either register the context during startup
+        // TODO: or use `getStartServices()` here (will force to implement getStartService for internalCore though...)
         const savedObjectsClient = coreSetup.savedObjects.getScopedClient(req);
         const uiSettingsClient = coreSetup.uiSettings.asScopedToClient(savedObjectsClient);
```

Comment:
This is a pain point. As of PR state, `server#registerCoreContext` now requires API from both core `setup` and `start`. Not sure what is the best way to address that.

Reply:
Yea, that's what I'm doing now. The handled is not invoked before start anyway, so this should be totally safe.